src/hooks/use-load-image.js

```diff
@@ -0,0 +1,33 @@
+import { usePromise } from "@nebula.js/stardust";
+
+// Create a promise that ensures the button background image is loaded properly
+const imagePromise = (url) =>
+  new Promise((resolve, reject) => {
+    const img = new Image();
+    img.src = url;
```

Comment:
In style formatter uses `getImageUrl` to edit the url, do you need to do similar here?